RecoEcal/EgammaClusterProducers/src/SCEnergyCorrectorDRNProducer.cc

```diff
@@ -0,0 +1,111 @@
+#include "HeterogeneousCore/SonicTriton/interface/TritonEDProducer.h"
+#include "HeterogeneousCore/SonicTriton/interface/TritonData.h"
+
+#include "FWCore/Framework/interface/stream/EDProducer.h"
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+
+#include "DataFormats/EgammaReco/interface/SuperCluster.h"
+#include "DataFormats/EgammaReco/interface/SuperClusterFwd.h"
+#include "DataFormats/Common/interface/ValueMap.h"
+
+#include "DataFormats/EgammaCandidates/interface/GsfElectron.h"
+
+#include "RecoEcal/EgammaClusterAlgos/interface/SCEnergyCorrectorDRN.h"
+
+#include <sstream>
+#include <string>
+#include <vector>
+#include <random>
+
+/*
+ * SCEnergyCorrectorDRNProducer
+ *
+ * Simple producer to generate a set of corrected superclusters with the DRN regression
+ * Based on RecoEcal/EgammaClusterProducers/SCEnergyCorrectorProducer by S. Harper (RAL/CERN)
+ *
+ * Author: Simon Rothman (UMN, MIT)
+ *
+ */
+
+static float sigmoid(float x) { return 1.0f / (1.0f + exp(-x)); }
+
+static float logcorrection(float x) {
+  static float ln2 = log(2);
+  return ln2 * 2 * (sigmoid(x) - 0.5);
+}
+
+static float correction(float x) { return exp(-logcorrection(x)); }
```

Comment:
Are you going to call these anywhere outside the class? Perhaps it makes sense to make these static class methods instead?

Reply:
These are just quick utility functions only needed in this producer. My understanding is that declaring them as static functions restricts their usage to this .cc file, which strikes me as the cleanest way to keep them from cluttering up any other namespace.

Reply:
The preferred way to restrict functions to a cc file is an [anonymous namespace](https://en.cppreference.com/w/cpp/language/namespace#Unnamed_namespaces).

Reply:
Okay, I will do this instead. Thanks